sourcemash/api/items.py

```diff
@@ -20,16 +22,44 @@
     'uri': fields.Url('api.item')
 }
 
-class ItemListAPI(Resource):
-
-    def get(self, feed_id):
-        return {'items': [marshal(item, item_fields) for item in Item.query.filter_by(feed_id=feed_id).all()]}
 
 class ItemAPI(Resource):
 
-    def get(self, feed_id, id):
+    def get(self, id):
         item = Item.query.get_or_404(id)
         return {'item': marshal(item, item_fields)}
 
-api.add_resource(ItemListAPI, '/feeds/<int:feed_id>/items', endpoint='items')
-api.add_resource(ItemAPI, '/feeds/<int:feed_id>/items/<int:id>', endpoint='item')
\ No newline at end of file
+
+class FeedItemListAPI(Resource):
+
+    def get(self, feed_id):
+        return {'items': [marshal(item, item_fields) for item in Item.query.filter_by(feed_id=feed_id).all()]}
+
+
+class CategoryItemListAPI(Resource):
+
+    @login_required
+    def get(self, category):
+        category = category.title()
+        user_feed_ids = [feed.id for feed in current_user.subscribed]
+
+        items = Item.query.filter((Item.category_1 == category) | (Item.category_2 == category))    \
+                            .filter(Item.feed_id.in_(user_feed_ids))                                \
+                            .all()
+
+        return {'items': [marshal(item, item_fields) for item in items]}
+
+
+class CategoryItemListAllAPI(Resource):
+
+    def get(self, category):
+        category = category.title()
+        items = Item.query.filter((Item.category_1 == category) | (Item.category_2 == category)).all()
+        return {'items': [marshal(item, item_fields) for item in items]}
+
+
+api.add_resource(ItemAPI, '/items/<int:id>', endpoint='item')
+api.add_resource(FeedItemListAPI, '/feeds/<int:feed_id>/items', endpoint='feed_items')
+api.add_resource(CategoryItemListAPI, '/categories/<string:category>/items', endpoint='category_items')
```

Comment:
The standard paths for RESTFUL is to have an id immediately follow a resource if you want one model in particular, and the resource you're handling (i.e. items) should be the last one listed so it's clearer. For the first point, an outside would be confused how we get categories from items without defining which item to look at.